js/reducers/homeReducer.js

```diff
@@ -15,6 +15,8 @@ function homeReducer(state = initialState, action) {
       });
     case MESSAGES_CLEAR:
       return Object.assign({}, state, { messages: [] });
+    case MESSAGES_SEARCH:
+      return Object.assign({}, state, { searchText: action.text });
```

Comment:
I mentioned something like this via chat, but I think we'll want to have a separate reducer for filter states.

Then we can take the list of messages and the list of applied filters, and derive the list of messages to show from that. I think eventually we'll probably want to use reselect to get the filtered list. If we want to make it easier for now, we could just use methods on the reducer. You can see an example of this in the Redux examples: https://github.com/reactjs/redux/blob/master/examples/shopping-cart/reducers/index.js

There are a few talks I'm not too interested in tomorrow, so we might have a chance to talk about it if you'd like.